### PR TITLE
added jade embedding support

### LINKED ISSUE #8: embed templates in jade
Jade has mixins. Make a mixin that makes it easy to embed templates. Pretty simple.

### PR DESCRIPTION
**Pull request by nelsonpecora**
fixes #8
- made the nunjucks `embed` filter super generic
- automatically add it as nunjucks filter
- automatically add it as jade option
- renamed all of this to `template.js`
- broke out generic tests into their own `describe` block
- added jade tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,6 +92,16 @@ The `data` you pass in is then used to render the child template. You can option
 
 Properties in the `data` object will overwrite properties of the same name in the `defaults` object, as this uses lodash's fast `_.defaults()` method.
 
+### Jade
+
+Jade embedding is very similar. Multiplex-templates adds an `embed` function to the options passed into any jade template, which takes three arguments, `data`, `name`, and `defaults`. The logic is the same as the nunjucks filter.
+
+```jade
```
Comment (cruzanmo) [inline README.md:99]: Is it possible to pass through defaults in jade like in the nunjucks example?

File: lib/engines.js [hunk 2]
```diff
@@ -15,7 +16,7 @@ _.map(supported, function(name) {
 });
 
 // add embed filters and mixins
-engines.nunjucks.addFilter('embed', require('./nunjucks').embedFilter);
+engines.nunjucks.addFilter('embed', template.embed);
```
Comment (cruzanmo) [inline lib/engines.js:19]: Does this need to run when using jade?

File: README.md (hunk 1 above)
Comment (nelsonpecora) [inline README.md:99]: Yeah? It says you can in the description...

File: lib/engines.js (hunk 2 above)
Comment (nelsonpecora) [inline lib/engines.js:19]: it instantiates both when it's required. I'm exploring alternatives in #2

File: README.md (hunk 1 above)
Comment (cruzanmo) [inline README.md:99]: But not in the example code snippet.

Comment (techmsi) [inline README.md:99]: Yes, we should add an example code snippet for that.

File: README.md [hunk 3]
```diff
@@ -92,6 +92,23 @@ The `data` you pass in is then used to render the child template. You can option
...
+
+Jade embedding is very similar. Multiplex-templates adds an `embed` function to the options passed into any jade template, which has the same logic as the nunjucks filter.
+
+```jade
+section#foo
+  p.embedded
+    != embed(data, 'name')
+```
+
+As with the nunjucks template, you can also pass a `defaults` object into it as a third argument.
```
Comment (techmsi) [inline README.md:105]: +1

**Comment by cruzanmo** [comment]
Looks good.

**Pull request merged**